wandb/integration/langchain/util.py

```diff
@@ -68,15 +68,9 @@ def safely_convert_model_to_dict(
     data = None
     message = None
     try:
-        data = model.dict()
+        data = model
```

Comment:
@tssweeney : Right now this is just the serialized dict passed by the `run_manager` -> `{"name": "LLMChain"}` etc. I noticed that you had serialized the chain/agent and passed it while monkey patching. This is not possible with the current `run_manager`. Would like appreciate some feedback/ideas here.

Reply:
yeah, this is a pain... basically without this stuff, then we don't get the model architecture feature. I have an idea, but it is equally as hacky.

Notice that a `__call__` method is:
https://github.com/hwchase17/langchain/blob/7e967aa4d581bec8b29e9ea44267505b0bad18b9/langchain/chains/base.py#LL125C1-L144C71

```
        inputs = self.prep_inputs(inputs)
        callback_manager = CallbackManager.configure(
            callbacks, self.callbacks, self.verbose
        )
        new_arg_supported = inspect.signature(self._call).parameters.get("run_manager")
        run_manager = callback_manager.on_chain_start(
            {"name": self.__class__.__name__},
            inputs,
        )
        try:
            outputs = (
                self._call(inputs, run_manager=run_manager)
                if new_arg_supported
                else self._call(inputs)
            )
        except (KeyboardInterrupt, Exception) as e:
            run_manager.on_chain_error(e)
            raise e
        run_manager.on_chain_end(outputs)
        return self.prep_outputs(inputs, outputs, return_only_outputs)
``` 

we could do something to "attach" `self` to the prepared inputs. In particular, imagine a patched `prepare_inputs` where we do something like:

```
def new_prepare_input(self, *args, **kwargs):
   res = old_preprare_input(*args, **kwargs)
   # Attach `self` to ride along
   res._wb_monkeypatched_self = self # you can do this b/c python is silly
   return res
```

Then in our `safely_get_span_producing_model` code, we extract the `self` from inputs and serialize it.

Reply:
This would still require monkey patching langchain methods. Maybe we can add this in the next PR along with autologging.

Reply:
yes, this certainly would require monkeypatching - i am just trying to think of how we can save the model architecture feature.

Reply:
@tssweeney : Changed this to add monkey patched the __call__ method to enable logging the model architecture. What do you think ?

Reply:
I have removed monkeypatching per our slack convo